interview-questions.md

```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 2. What is a block in Ruby?
 
-  Your answer:
+  Your answer: These are chunks of code put together in a block and I believe that they are called with a function of the same name. It's like having a file cabinet full of files. You can open up the file drawer Labeled a through C and know that you are going to get files or blocks full of names a through C.
 
-  Researched answer:
+  Researched answer: In Ruby they have blocks which are like anonymous functions similar to JavaScript. Blocks can be created with the do/end keywords OR with {}. Blocks are code that is passed to a method. I always like to think of these as building blocks.Each building block contains code that makes it easier to stack on top of one another and make a tower like I used to do when I was five. They are just another tool in my arsenal I can use in my future career.
 
 
 
 3. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: I do believe an implicit return is a return that wasn't exactly asked for but Ruby has returned the last executable instructions. Meaning that Ruby doesn't really know what you're asking for but it will show you where it the code broke.
 
-  Researched answer:
+  Researched answer: So I was kind of right according to the syllabus it states that implicit return means the value of the last line of a method is automatically returned without using the keyword return. When using implicit returns, object literals must be wrapped in parenthesis so that the curly braces are not mistaken for the opening of the function’s body.
 
 
 
 4. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: Object oriented programming deals with building your project or code by using objects like building blocks and functional programming is like building the project or the code for functionality not structure wise.
 
-  Researched answer:
+  Researched answer:My research the definition of object oriented programming is s a programming paradigm that relies on the concept of classes and objects. It is used to structure a software program into simple, reusable pieces of code blueprints (usually called classes), which are used to create individual instances of objects. The definition of functional programming is s a programming paradigm that relies on the concept of classes and objects. It is used to structure a software program into simple, reusable pieces of code blueprints (usually called classes), which are used to create individual instances of objects. By reading these two definitions you can see that they are as different as No butter and jelly. They are good by themselves but they're really good when put together.
```

Comment:
I think you have some copy pasta issues on the third sentence through the 4th sentence.